storm-server/src/main/java/org/apache/storm/daemon/nimbus/Nimbus.java

```diff
@@ -4176,12 +4176,13 @@ private void maybeAddPlaceholderSpoutAggStats(TopologyPageInfo topoPageInfo, Sto
      *
      * @param topoPageInfo  topology page info holding bolt AggStats
      * @param topology      storm topology used to get bolt names
+     * @param includeSys    whether to show system bolts
      */
-    private void maybeAddPlaceholderBoltAggStats(TopologyPageInfo topoPageInfo, StormTopology topology) {
+    private void maybeAddPlaceholderBoltAggStats(TopologyPageInfo topoPageInfo, StormTopology topology, boolean includeSys) {
         if (topoPageInfo.get_id_to_bolt_agg_stats().isEmpty()) {
             Map<String, Bolt> bolts = topology.get_bolts();
-            for (Entry<String, Bolt> entry : bolts.entrySet()) {
-                if (Utils.isSystemId(entry.getKey())) {
+            for (String boltName : bolts.keySet()) {
+                if ((!includeSys && Utils.isSystemId(boltName)) || boltName.equals("__system")) {
```

Comment:
`Constants.SYSTEM_COMPONENT_ID` Is a better instead of `__system`

Reply:
Changed to use Constants.SYSTEM_COMPONENT_ID